enf-reducer/promote-common-constraints.metta

```diff
@@ -0,0 +1,68 @@
+;(: promoteCommonConstraints (-> Expression Expression))
+(= (promoteCommonConstraints $expr)
+    (let* 
+        (
+            (($literals $children) (getGsetAndChildren $expr)) 
+            ($childrenLiteral   (collapse (getGuardSet (superpose $children) ())))
+            ($shared (findCommon $childrenLiteral))
+            ($common (findIntersection $literals $shared))   
+            ($x 2)      
+        )
+        (if (== $common ())
+            (() $expr)
+            (let*
+                ( 
+                ($updatedChildren (removeCommonLiterals $common $children))
+                ($updatedLiterals (removeElement $common $literals))
+                ($newExp (concatTuple $updatedLiterals $updatedChildren ))
+                ($res (cons-atom OR $newExp))              
```

Comment:
I believe this transformation rule is wrote with the assumption that it only accepts an `OR` node right?

Reply:
yeah, The POA will determined by the rte, so we can assume `$expr` is `OR `node